src/components/analyses/listing/Listing.js

```diff
@@ -542,45 +490,95 @@ function Listing(props) {
     };
 
     const handleChangePage = (event, newPage) => {
-        setPage(newPage - 1);
-        setSelected([]);
+        onRouteToListing &&
+            onRouteToListing(
+                order,
+                orderBy,
+                newPage - 1,
+                rowsPerPage,
+                permFilter,
+                typeFilter,
+                idFilter
+            );
     };
 
     const handleChangeRowsPerPage = (newPageSize) => {
-        setRowsPerPage(parseInt(newPageSize, 10));
-        setSelected([]);
-        setPage(0);
+        onRouteToListing &&
+            onRouteToListing(
+                order,
+                orderBy,
+                0,
+                parseInt(newPageSize, 10),
+                permFilter,
+                typeFilter,
+                idFilter
+            );
     };
     const handleRequestSort = (event, property) => {
-        const isAsc = orderBy === property && order === "asc";
-        setOrder(isAsc ? "desc" : "asc");
-        setOrderBy(property);
-        setSelected([]);
-        setPage(0);
+        const isAsc =
+            orderBy === property && order === globalConstants.SORT_ASCENDING;
+        onRouteToListing &&
+            onRouteToListing(
+                isAsc
+                    ? globalConstants.SORT_DESCENDING
+                    : globalConstants.SORT_ASCENDING,
+                property,
+                page,
+                rowsPerPage,
+                permFilter,
+                typeFilter,
+                idFilter
+            );
     };
 
     const handleAppTypeFilterChange = (appTypeFilter) => {
-        setAppTypeFilter(appTypeFilter);
         setSelected([]);
-        setPage(0);
+        onRouteToListing &&
+            onRouteToListing(
+                order,
+                orderBy,
+                0,
+                rowsPerPage,
+                permFilter,
+                appTypeFilter,
+                idFilter
+            );
     };
 
     const handleOwnershipFilterChange = (viewFilter) => {
-        setPermFilter(viewFilter);
         setSelected([]);
-        setPage(0);
+        onRouteToListing &&
+            onRouteToListing(
+                order,
+                orderBy,
+                0,
+                rowsPerPage,
+                viewFilter,
+                typeFilter,
+                idFilter
+            );
     };
 
     const handleBatchIconClick = (analysis) => {
         setParentAnalyses(analysis);
-        setPermFilter(null);
-        setAppTypeFilter(null);
-        setIdFilter(null);
         setSelected([]);
+        onRouteToListing &&
+            onRouteToListing(
+                order,
+                orderBy,
+                0,
+                rowsPerPage,
+                null,
+                null,
+                idFilter
```

Comment:
The `setIdFilter(null)` call was in this function before, so should `idFilter` be `null` in this `onRouteToListing` now?
```suggestion
                null
```
I'm not really sure about this handler now. It seems like it's setting `parentAnalyses` state, but then it's navigating the page away with `onRouteToListing`. So will the `parentAnalyses` state be preserved after the route navigation?

Reply:
At least thats my understanding. The components are not unmounted util you route to a different page.